crates/claims/crates/vc/src/data_model/credential.rs

```diff
@@ -0,0 +1,224 @@
+    /// supporting information in a verifiable credential.
+    type Evidence: Evidence;
+
+    /// Credential Schemas (Zero-Knowledge Proofs).
+    ///
+    /// See: <https://www.w3.org/TR/vc-data-model//#zero-knowledge-proofs>
+    type Schema;
+
+    /// Identifier.
+    fn id(&self) -> Option<&Uri> {
+        None
+    }
+
+    /// Types that are **not** `VerifiableCredential`.
+    ///
+    /// Since the `VerifiableCredential` type is *required*, it is omitted from
+    /// the value returned by this function. If you need to iterate over
+    /// all the credential types, including `VerifiableCredential`, use the
+    /// [`Self::types`] method.
+    fn additional_types(&self) -> &[String] {
+        &[]
+    }
+
+    fn types(&self) -> CredentialTypes {
+        CredentialTypes {
+            base_type: true,
+            additional_types: self.additional_types().iter(),
+        }
+    }
+
+    /// Credential subject.
+    fn credential_subjects(&self) -> &[Self::Subject] {
+        &[]
+    }
+
+    /// Issuer.
+    ///
+    /// This property is *required* for the credential to be *verifiable*.
+    fn issuer(&self) -> &Self::Issuer;
+
+    /// Issuance date.
+    fn issuance_date(&self) -> DateTime;
+
+    /// Expiration date.
+    fn expiration_date(&self) -> Option<DateTime> {
+        None
+    }
+
+    /// Credential status.
+    ///
+    /// Used for discovery of information about the current status of a
+    /// verifiable credential, such as whether it is suspended or revoked.
+    fn credential_status(&self) -> &[Self::Status] {
+        &[]
+    }
+
+    /// Refresh services.
+    ///
+    /// See: <https://www.w3.org/TR/vc-data-model//#refreshing>
+    fn refresh_services(&self) -> &[Self::RefreshService] {
+        &[]
+    }
+
+    /// Terms of Use.
+    ///
+    /// Terms of use can be utilized by an issuer or a holder to communicate the
+    /// terms under which a verifiable credential or verifiable presentation was
+    /// issued.
+    fn terms_of_use(&self) -> &[Self::TermsOfUse] {
+        &[]
+    }
+
+    /// Evidences.
+    ///
+    /// Can be included by an issuer to provide the verifier with additional
+    /// supporting information in a verifiable credential.
+    fn evidences(&self) -> &[Self::Evidence] {
+        &[]
+    }
+
+    /// Credential Schemas (Zero-Knowledge Proofs).
+    ///
+    /// See: <https://www.w3.org/TR/vc-data-model//#zero-knowledge-proofs>
+    fn credential_schemas(&self) -> &[Self::Schema] {
+        &[]
+    }
+
+    /// Validates the credential.
+    ///
+    /// Validation consists in verifying that the claims themselves are
+    /// consistent and valid with regard to the verification environment.
+    /// For instance, checking that a credential's expiration date is not in the
+    /// past, or the issue date not in the future.
+    ///
+    /// Validation may fail even if the credential proof is successfully
+    /// verified.
+    fn is_valid_credential(&self) -> bool {
```

Comment:
I agree, validation deserves a bit more work, but should I refine it in this PR? For instance, besides a proper `Result` output type, it would be nice to have an input argument providing some kind of environment used to validate some claims such as the expiration date (requires the current date), the intended audience (requires the verifier's domain name), etc.

Reply:
Yeah, and also this is where the distinction between validation and verification becomes fuzzy. Considering making an error type that would work for all credential would be very difficult, I would say maybe change to `fn validate(&self) -> Result<(), String>` in this PR and that should be good enough for a while